feat: Add frontend development overview and useful resources (#56)
shesharpnl/knowledge-hub: resources/frontend-development.md

# Frontend Development

Frontend Development is a pivotal aspect of web development that focuses on crafting the visual and interactive components of a web application. It encompasses two fundamental elements: the graphical design and the user interface (UI) development.

## Graphical Design

> In the context of frontend development, graphical design involves conceptualizing and creating the visually appealing layouts, typography,color schemes, and graphical elements that collectively form the aesthetic foundation of the application.Designers employ tools like Adobe Photoshop, Sketch, or Figma to translate ideas into tangible visual assets.

Reviewer: Just some small type changes for readability here:

`In the context of frontend development, graphical design involves conceptualizing and creating the visually appealing layouts, typography, color schemes, and graphical elements that collectively form the aesthetic foundation of the application. Designers employ tools like Adobe Photoshop, Sketch, or Figma to translate ideas into tangible visual assets.`

Author: Done. Thank you for the review, @mariahlaqua.

## User Interface (UI) Development

> UI development focuses on implementing the graphical design into functional, interactive interfaces that users can seamlessly navigate and engage with. It includes writing codes using technologies like HTML, CSS, and JavaScript to structure content, style elements, and add dynamic behavior.

Author: Done.

> ### Key Aspects of Frontend Development

Author: Done.

#### HTML (HyperText Markup Language)

This is a markup language used to create the structure and layout of web pages.

#### CSS (Cascading Style Sheets)

> It controls the presentation, formatting, and appearance of web pages, such as colors, fonts, and layout.

Author: Done.

> #### JavaScript

Reviewer 2: There is a JavaScript section starts from line 53.

Same as the CSS, if there is no reason for duplication, let's move them here.

Author: Done.

Enables dynamic interactions and functionality on web pages, such as form validation, animations, and data manipulation. JavaScript frameworks and libraries like React, Vue.js, and Angular facilitate the development of complex user interfaces and enhance interactivity.

A list of useful websites to gain mastery of frontend development:

> - [Frontend Masters](https://frontendmasters.com/):
> 
> Frontend Masters is a comprehensive learning platform for frontend developers, offering in-depth courses on a wide range of topics, including HTML, CSS, JavaScript, and modern frontend frameworks. With expert-led instruction and hands-on projects, developers can enhance their skills and stay up-to-date with the latest trends in frontend development.

Reviewer 2: ```suggestion
- **[Frontend Masters](https://frontendmasters.com/)**

Frontend Masters is a comprehensive learning platform for frontend developers, offering in-depth courses on a wide range of topics, including HTML, CSS, JavaScript, and modern frontend frameworks. With expert-led instruction and hands-on projects, developers can enhance their skills and stay up-to-date with the latest trends in frontend development.
```

Author: Done.

- [The Odin Project](https://www.theodinproject.com/):

> The Odin Project provides a free and open-source curriculum for learning web development, covering HTML, CSS, JavaScript, and other essential technologies. Through project-based learning and guided tutorials, learners can build real-world projects and gain practical experience in frontend development, preparing them for careers in the tech industry.

Reviewer 2: Let's move these resources to right after the title of "Resources to Learn Frontend Development".

Author: Done.

> - [The Odin Project](https://www.theodinproject.com/):
> 
> The Odin Project provides a free and open-source curriculum for learning web development, covering HTML, CSS, JavaScript, and other essential technologies. Through project-based learning and guided tutorials, learners can build real-world projects and gain practical experience in frontend development, preparing them for careers in the tech industry.

Reviewer 2: ```suggestion
- **[The Odin Project](https://www.theodinproject.com/)**

The Odin Project is an open source that provides free curriculum for learning web development, covering HTML, CSS, JavaScript, and other essential technologies. Through project-based learning and guided tutorials, learners can build real-world projects and gain practical experience in frontend development, preparing them for careers in the tech industry.
```

Author: Done.

## CSS

Here is a list of useful websites for learning and mastering CSS:

> - [Codecademy](https://www.codecademy.com/learn/learn-css) -[freeCodeCamp](https://www.freecodecamp.org/learn/)
> 
> Codecademy offers an interactive platform for learning CSS, providing hands-on exercises and projects to reinforce concepts. From basic syntax to advanced styling techniques, this course guides learners through the essentials of Cascading Style Sheets, empowering them to create visually appealing web designs.

Reviewer 2: 1. There are two lists in one line here. Let's fix them.
2. Please give brief description for the freeCodeCamp CSS resource.
3. Please fix the format for all list with description in this page.
      
     See the format for the Codecademy here as example:
       
       1. Give a space after dash.
       2. Add double asterisks to bold the title.
       3. Add space after the title.
       4. Add three space by hitting the spacebar before writing the description.
       
    You can use [Markdown Live Preview](https://markdownlivepreview.com/) to see the format result of Markdown.

```suggestion
- **[Codecademy](https://www.codecademy.com/learn/learn-css)**
   
   Codecademy offers an interactive platform for learning CSS, providing hands-on exercises and projects to reinforce concepts. From basic syntax to advanced styling techniques, this course guides learners through the essentials of Cascading Style Sheets, empowering them to create visually appealing web designs.

- **[freeCodeCamp](https://www.freecodecamp.org/learn/)**
```

Author: Done.

> - [Flexbox Frogy](https://flexboxfroggy.com/)

Reviewer 2: ```suggestion
- **[Flexbox Frogy](https://flexboxfroggy.com/)**
```

> Explore Flexbox Froggy, an interactive game that teaches Flexbox layout techniques through gamified challenges. By guiding frogs to their lily pads using Flexbox properties, users gain a hands-on understanding of Flexbox's powerful capabilities for building responsive web layouts.

Reviewer 2: ```suggestion
Explore Flexbox Froggy, an interactive game that teaches CSS flexbox layout techniques through gamified challenges. By guiding frogs to their lily pads using flexbox properties, users gain a hands-on understanding of flexbox's powerful capabilities for building responsive web layouts.
```

Author: Done.

> -[CSS Grid Garden](https://cssgridgarden.com/)

Reviewer 2: 1. Add a space after dash.
2. Bold the title.

```suggestion
- **[CSS Grid Garden](https://cssgridgarden.com/)**
```

> CSS Grid Garden offers an engaging way to learn CSS Grid Layout through a series of garden-themed challenges. By planting and arranging trees using CSS Grid properties, participants master the intricacies of grid-based layouts and gain confidence in designing complex web interfaces.

Reviewer 2: ```suggestion
CSS Grid Garden offers an engaging way to learn CSS grid through a series of garden-themed challenges. By planting and arranging trees using grid properties, participants master the intricacies of grid-based layouts and gain confidence in designing complex web interfaces.
```

Author: Done.

## JavaScript

JavaScript is a scripting or programming language that allows you to create dynamic web pages.

A list of useful resources for learning JavaScript:

> - [MDN Web Docs - JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript)

Reviewer 2: ```suggestion
- **[MDN Web Docs - JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript)**
```

Author: Done.

MDN Web Docs offers an extensive resource on JavaScript, providing detailed documentation, tutorials, and references for developers. From basic syntax explanations to advanced topics like asynchronous programming, developers can rely on MDN for accurate and up-to-date information on JavaScript features and APIs.

### JavaScript Library

> A javaScript library is a toolbox of pre-written code designed for a specific task. In other words,it's a collection of reusable functions and features framed or crafted to make certain programming tasks easier. For a beginner, think of it as a set of tools that you can use to build a web application without having to create everything from scratch. An example of a JavaScript library is React.

Reviewer 2: ```suggestion
A JavaScript library is a toolbox of pre-written code designed for a specific task. In other words, it's a collection of reusable functions and features framed or crafted to make certain programming tasks easier. For beginners, think of it as a set of tools that you can use to build a web application without having to create everything from scratch. An example of a JavaScript library is React.
```

Author: Done.
